core/vm/actor/version.hpp

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <type_traits>
+
+namespace fc::vm::actor {
+  enum class ActorVersion {
+    kVersion0 = 0,
+    kVersion2 = 2,
+    kVersion3 = 3,
+    kVersion4 = 4,
+    kVersion5 = 5,
+  };
+
+  struct WithActorVersion {
```

Comment:
I suggest add protected destructor, so it is impossible to delete derived object via base class. I urge to follow [guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#discussion-make-base-class-destructors-public-and-virtual-or-protected-and-non-virtual).

Reply:
No